server/index.js

```diff
@@ -23,103 +23,128 @@ const open = require('open');
 const portfinder = require('portfinder');
 const crypto = require("crypto");
 const fse = require('fs-extra');
+const { createHttpTerminator } = require('http-terminator');
 
+const DEFAULT_PORT = '9000';
 
-async function start(portPreferred) {
-    let port = process.env.ASSET_COMPUTE_DEV_PORT || '9000';
-    if (!isNaN(portPreferred)) {
-        port = portPreferred;
+class DevtoolServer {
+
+    async run (portPreferred) {
+        this.port = process.env.ASSET_COMPUTE_DEV_PORT || DEFAULT_PORT;
+        if (!isNaN(portPreferred)) {
```

Comment:
[isNan()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/isNaN) is a different function than [Number.isNaN()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isNaN). The second is newer and more robust, I'd prefer using `Number.isNaN()`.

Reply:
this does not work:
<img width="255" alt="Screen Shot 2020-09-16 at 9 18 24 AM" src="https://user-images.githubusercontent.com/20048485/93364584-93ff7380-f7fd-11ea-8f70-8e724c568586.png">

`undefined` which is the default for `portPreferred` will always set that as the port value

Reply:
Ok - that was the reason then. Got it.